### DISCUSSION TITLE
Cannot delete or republish

### DISCUSSION THREAD

**Discussion by hktalent** [npm/feedback#1023] [CLI]
I once released a package, but later a bug was discovered and I needed to re-release it. However, there were too many projects used and I couldn't release it by increasing the version. I tried to cancel the release, delete the release, and tried to overwrite the release, but it always failed. , what you get is: Cannot publish over previously published version "2.2.1"
Because I tried releasing a new version, but couldn't get it processed in several projects
So I will back out to version 2.2.2 and try to overwrite version 2.2.1, but to no avail.

**Comment by ljharb** [comment]
A version is immutable, so the most you can do is publish v2.2.2 and deprecate v2.2.1.

**Reply by hktalent** [reply]
@ljharb Thank you very much for your enthusiastic answer. The main reason is that there are too many projects and it is painful to modify. Therefore, I want to cover the release. I suggest optimizing this mechanism, even if it is provided to pro users.
Looking forward to hearing from you
Thank you very much

**Reply by ljharb** [reply]
It wouldn't be an optimization, it would be disastrous and would destroy many of the guarantees the word relies on.

If v2.2.1 is not ideal, your ONLY OPTION, with no exceptions, is to publish a v2.2.2 and deprecate v2.2.1. v2.2.1 will exist forever and nothing will ever change that.

**Reply by hktalent** [reply]
@ljharb Are these your servers?
```
....
```
Your server seems to have some security issues
It has been reported to hackerone and is still under investigation.

**Reply by ljharb** [reply]
I’m not part of npm, and discussing security issues publicly would be irresponsible disclosure.

**Reply by hktalent** [reply]
All right
